components/client.jsx

```diff
@@ -27,13 +26,18 @@ function createElement(Component, props) {
 
 function onEnter(nextState, replaceState) {
   var result = redirectURLParser(nextState.location);
-  if (result.pathname) {
-    replaceState({}, result.pathname, result.query);
+  if (!result.pathname) {
+    return;
+  }
+  if (result.query) {
+    replaceState({pathname: result.pathname, query: result.query});
+  } else {
+    replaceState(result.pathname);
```

Comment:
replaceState has a new function signature now. It won't accept pathname with a built in query string if it's passed in as an object, like in line 33, so now I need to make the distinction between the two.